sycl/include/CL/sycl/detail/image_accessor_util.hpp

```diff
@@ -855,6 +860,88 @@ DataTy ReadPixelDataNearestFiltMode(cl_int4 PixelCoord, id<3> ImgPitch,
+  DataT RetData;
+  if (isOutOfRange(PixelCoord, SmplAddrMode, ImgRange)) {
+    cl_float4 BorderColor = (getBorderColor(ImgChannelOrder));
+    RetData = BorderColor.convert<typename TryToGetElementType<DataT>::type>();
+  } else {
+    RetData = ReadPixelData<DataT>(PixelCoord, ImgPitch, ImgChannelType,
+                                   ImgChannelOrder, BasePtr, ElementSize);
+  }
+  return RetData;
+}
+
+// Computes and returns color value with Linear Filter Mode.
+// Steps:
+// 1. Computes the 8 coordinates using all combinations of i0/i1,j0/j1,k0/k1.
+// 2. Calls getColor() on each Coordinate.(Ci*j*k*)
+// 3. Computes the return Color Value using a,b,c and the Color values. 
+template <typename DataT>
+DataT ReadPixelDataLinearFiltMode(cl_int8 CoordValues, cl_float4 abc,
+                                  addressing_mode SmplAddrMode,
+                                  range<3> ImgRange, id<3> ImgPitch,
+                                  image_channel_type ImgChannelType,
+                                  image_channel_order ImgChannelOrder,
+                                  void *BasePtr, uint8_t ElementSize) {
+  cl_int i0 = CoordValues.s0(), j0 = CoordValues.s1(), k0 = CoordValues.s2(),
+         i1 = CoordValues.s4(), j1 = CoordValues.s5(), k1 = CoordValues.s6();
+  cl_float a = abc.x();
+  cl_float b = abc.y();
+  cl_float c = abc.z();
+  cl_float4 RetData;
+
+  // Get Color Values at each Coordinate.
+  DataT Ci0j0k0 =
+      getColor<DataT>(cl_int4{i0, j0, k0, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+  DataT Ci1j0k0 =
+      getColor<DataT>(cl_int4{i1, j0, k0, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+  DataT Ci0j1k0 =
+      getColor<DataT>(cl_int4{i0, j1, k0, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+  DataT Ci1j1k0 =
+      getColor<DataT>(cl_int4{i1, j1, k0, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+  DataT Ci0j0k1 =
+      getColor<DataT>(cl_int4{i0, j0, k1, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+  DataT Ci1j0k1 =
+      getColor<DataT>(cl_int4{i1, j0, k1, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+  DataT Ci0j1k1 =
+      getColor<DataT>(cl_int4{i0, j1, k1, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+  DataT Ci1j1k1 =
+      getColor<DataT>(cl_int4{i1, j1, k1, 0}, SmplAddrMode, ImgRange, ImgPitch,
+                      ImgChannelType, ImgChannelOrder, BasePtr, ElementSize);
+
+  RetData =
+      ((1 - a) * (1 - b) * (1 - c) * Ci0j0k0.template convert<cl_float>()) +
```

Comment:
please break down this formula into intermediate calculations to make it easier to read.

Reply:
You have missed some comments. Please take a look.

Reply:
sorry about that.. I will do the needed changes.

Reply:
done.